instrumentation/metric/registry.go

```diff
@@ -80,33 +79,29 @@ func (r *inMemoryRegistry) WithNodeAddress(nodeAddress primitives.NodeAddress) R
 func (r *inMemoryRegistry) Get(metricName string) metric {
 	r.mu.RLock()
 	defer r.mu.RUnlock()
-	for _, metric := range r.mu.metrics {
-		if metric.Name() == metricName {
-			return metric
-		}
-	}
-	return nil
+	return r.mu.metrics[metricName]
 }
 
+// Only if the actual metric (the object/pointer) is the same remove metric
```

Comment:
would be nice if we would return two booleans - or a tristate:
1. metric was found
2. metric was removed

but if it's not required no need,... just a thought